src/shard_window_ratio.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+#pragma once
+
+#include <stddef.h>
+#include <math.h>
+#include <stdbool.h>
+#include <string.h>
+#include "redismodule.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Check if a command is a shard-level command (_FT.*) vs coordinator command (FT.*).
+ *
+ * @param argv Command arguments array
+ * @param argc Number of arguments
+ * @return true if this is a shard-level command, false otherwise
+ */
+static inline bool isShardLevelCommand(RedisModuleString **argv, int argc) {
+  if (argc == 0 || argv == NULL) {
+    return false;
+  }
+
+  size_t cmdLen;
+  const char *cmdName = RedisModule_StringPtrLen(argv[0], &cmdLen);
+  return (cmdLen > 3 && cmdName[0] == '_' && strncasecmp(cmdName + 1, "FT.", 3) == 0);
```

Comment:
is there a better way?

Reply:
@raz-mon Is there some API to detect whether a command is an internal command (starting from redis>= 8.0)?

Reply:
It's currently checked similarly [here](https://github.com/RediSearch/RediSearch/blob/b1332489e7dde8535c3774201e4ad1652b4f08d8/src/aggregate/aggregate_exec.c#L937-L938), and a flag is turned on.
But the term "internal" is abused a bit, since we have underscored commands that are not internal (e.g., `_FT.CONFIG`) in the sense that they are not disabled for non-internal connections.
Anyways, it seems that the check here is for shard commands rather than internal commands, for which we have the `QEXEC_F_INTERNAL` flag but no dedicated API (would be nice to have).